ibm-demo: db2 jdbc connect did not work.

DB2 connector failed after installing via
make connectdb2source

Error in connect logs:
INFO Unable to connect to database on attempt 1/3. Will retry in 10000 ms. (io.confluent.connect.jdbc.util.CachedConnectionProvider)
com.ibm.db2.jcc.am.DisconnectNonTransientConnectionException: [jcc][t4][2043][11550][4.29.24] Exception java.net.ConnectException: Error opening socket to server ibmdb2/172.18.0.3 on port 50,000 with message: Connection refused (Connection refused). ERRORCODE=-4499, SQLSTATE=08001
        at com.ibm.db2.jcc.am.b7.a(b7.java:338)
        at com.ibm.db2.jcc.am.b7.a(b7.java:435)
        at com.ibm.db2.jcc.t4.a0.a(a0.java:462)
        at com.ibm.db2.jcc.t4.a0.<init>(a0.java:97)
        at com.ibm.db2.jcc.t4.a.b(a.java:367)
        at com.ibm.db2.jcc.t4.b.newAgent_(b.java:2251)
        at com.ibm.db2.jcc.am.Connection.initConnection(Connection.java:855)
        at com.ibm.db2.jcc.am.Connection.<init>(Connection.java:800)
        at com.ibm.db2.jcc.t4.b.<init>(b.java:364)
        at com.ibm.db2.jcc.DB2SimpleDataSource.getConnection(DB2SimpleDataSource.java:243)
        at com.ibm.db2.jcc.DB2SimpleDataSource.getConnection(DB2SimpleDataSource.java:200)
        at com.ibm.db2.jcc.DB2Driver.connect(DB2Driver.java:491)
        at com.ibm.db2.jcc.DB2Driver.connect(DB2Driver.java:117)
        at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
        at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:189)
        at io.confluent.connect.jdbc.dialect.GenericDatabaseDialect.getConnection(GenericDatabaseDialect.java:234)
        at io.confluent.connect.jdbc.util.CachedConnectionProvider.newConnection(CachedConnectionProvider.java:83)
        at io.confluent.connect.jdbc.util.CachedConnectionProvider.getConnection(CachedConnectionProvider.java:52)
        at io.confluent.connect.jdbc.JdbcSourceConnector.start(JdbcSourceConnector.java:92)
        at org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:186)
        at org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:211)
        at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:350)
        at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:333)
        at org.apache.kafka.connect.runtime.WorkerConnector.doRun(WorkerConnector.java:141)
        at org.apache.kafka.connect.runtime.WorkerConnector.run(WorkerConnector.java:118)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
        at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
        at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
        at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.base/java.net.Socket.connect(Socket.java:609)
        at com.ibm.db2.jcc.t4.x.run(x.java:49)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at com.ibm.db2.jcc.t4.a0.a(a0.java:444)

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -5,10 +5,14 @@
 This repository demonstrates how to integrate with IBM technologies (IBM MQ and DB2). Two connectors will be started up: Datagen source, to mock clickstream data and IBM MQ Connetor source. Then we'll use KSQL to join the two sources together. We'll also configure a IBM DB2 source connector to read data from DB2. The resut of the ksqlDB join will be sent to IBM MQ using a sink connector.
 
 ## Download the demo
-Clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
+Using your terminal, clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
 
 ```bash
 git clone https://github.com/confluentinc/demo-scene
+```
+Then enter the demo folder:
+
+```bash
 cd demo-scene/ibm-demo
 ```
 

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -5,11 +5,18 @@
 This repository demonstrates how to integrate with IBM technologies (IBM MQ and DB2). Two connectors will be started up: Datagen source, to mock clickstream data and IBM MQ Connetor source. Then we'll use KSQL to join the two sources together. We'll also configure a IBM DB2 source connector to read data from DB2. The resut of the ksqlDB join will be sent to IBM MQ using a sink connector.
 
 ## Download the demo
-Using your terminal, clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
+Using your terminal, Download the zip containing this [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
 
 ```bash
-git clone https://github.com/confluentinc/demo-scene
-cd demo-scene/ibm-demo
+wget http://github.com/confluentinc/demo-scene/archive/master.zip
+```
+
+Then unzip the file and enter in the directory demo-scene-master/ibm-demo from your teminal.
+If you are using a Mac or similar commands should be:
+
+```bash
+unzip master.zip
+cd demo-scene-master/ibm-demo
 ```
 
 ## Make commands

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -5,14 +5,10 @@
 This repository demonstrates how to integrate with IBM technologies (IBM MQ and DB2). Two connectors will be started up: Datagen source, to mock clickstream data and IBM MQ Connetor source. Then we'll use KSQL to join the two sources together. We'll also configure a IBM DB2 source connector to read data from DB2. The resut of the ksqlDB join will be sent to IBM MQ using a sink connector.
 
 ## Download the demo
-Using your terminal, clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
+Clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
 
 ```bash
 git clone https://github.com/confluentinc/demo-scene
-```
-Then enter the demo folder:
-
-```bash
 cd demo-scene/ibm-demo
 ```
 

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -5,7 +5,7 @@
 This repository demonstrates how to integrate with IBM technologies (IBM MQ and DB2). Two connectors will be started up: Datagen source, to mock clickstream data and IBM MQ Connetor source. Then we'll use KSQL to join the two sources together. We'll also configure a IBM DB2 source connector to read data from DB2. The resut of the ksqlDB join will be sent to IBM MQ using a sink connector.
 
 ## Download the demo
-Using your terminal, lone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
+Using your terminal, clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
 
 ```bash
 git clone https://github.com/confluentinc/demo-scene

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -5,7 +5,7 @@
 This repository demonstrates how to integrate with IBM technologies (IBM MQ and DB2). Two connectors will be started up: Datagen source, to mock clickstream data and IBM MQ Connetor source. Then we'll use KSQL to join the two sources together. We'll also configure a IBM DB2 source connector to read data from DB2. The resut of the ksqlDB join will be sent to IBM MQ using a sink connector.
 
 ## Download the demo
-Clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
+Using your terminal, lone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
 
 ```bash
 git clone https://github.com/confluentinc/demo-scene

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -5,9 +5,12 @@
 This repository demonstrates how to integrate with IBM technologies (IBM MQ and DB2). Two connectors will be started up: Datagen source, to mock clickstream data and IBM MQ Connetor source. Then we'll use KSQL to join the two sources together. We'll also configure a IBM DB2 source connector to read data from DB2. The resut of the ksqlDB join will be sent to IBM MQ using a sink connector.
 
 ## Download the demo
-You can download the demo [here](https://bit.ly/3ex1tLx)
+Clone the [confluentinc/demo-scene](https://github.com/confluentinc/demo-scene) GitHub repository..
 
-Unzip the ibm-demo.zip and cd into the directory from your terminal.
+```bash
+git clone https://github.com/confluentinc/demo-scene
+cd demo-scene/ibm-demo
+```
 
 ## Make commands
 

--- a/ibm-demo/README.md
+++ b/ibm-demo/README.md
@@ -188,3 +188,14 @@ When you are done with the demo execute the command:
 ```conf
 make down
 ```
+
+## Troubleshooting tips
+
+```bash
+docker exec -ti ibmdb2 bash -c "su - db2inst1"
+db2 get dbm cfg | grep "SVCENAME"
+
+grep "db2c_db2inst1" /etc/services
+
+db2level
+```

--- a/ibm-demo/docker-compose.yml
+++ b/ibm-demo/docker-compose.yml
@@ -193,12 +193,12 @@ services:
       - .:/project
 
   ibmdb2:
-    image: ibmcom/db2
+    image: ibmcom/db2:11.5.6.0
     hostname: ibmdb2
     container_name: ibmdb2
     privileged: true
     ports:
-      - 50000:50000
+      - 25010:25010
     environment:
       LICENSE: accept
       DB2INST1_PASSWORD: passw0rd

--- a/ibm-demo/ibmmq/ibmdb2-source.json
+++ b/ibm-demo/ibmmq/ibmdb2-source.json
@@ -1,8 +1,7 @@
 {
 	"connector.class": "io.confluent.connect.jdbc.JdbcSourceConnector",
 	"tasks.max":"1",
-	"mq.hostname": "ibmmq",
-	"connection.url":"jdbc:db2://ibmdb2:50000/sample",
+	"connection.url":"jdbc:db2://ibmdb2:25010/sample",
 	"connection.user":"db2inst1",
 	"connection.password":"passw0rd",
 	"topic.prefix": "db2-",

